### ISSUE TITLE
Possible unintended usage of "info.stab" field

### ISSUE THREAD

**Issue by petrufm** [yasm/yasm#88]
Hi,

While experimenting with a CodeSonar plugin we develop, we noticed a potential bug in file "asm/modules/dbgfmts/stabs/stabs-dbgfmt.c" line 358 (actually in the chromium project that uses this project as a third-party).

yasm_error_set(YASM_ERROR_GENERAL,
                N_("stabs debugging conflicts with user-defined section .stabstr"));
yasm_errwarn_propagate(errwarns,
                yasm_section_bcs_first(info.stab)->line); //line 358

Shouldn't you use in line 358 info.stabstr (and not  info.stab)?

Thank you.

**Comment by MichaelUrman** [comment]
I agree that this looks incorrect. However, having not looked at this code in a decade, I'm hardly an expert anymore. On the bright side, it looks like the worst case scenario here is an incorrect line number in the error message, rather than use of uninitialized memory.

Do you have any test cases that could be applied here? They should be able to confirm the error and its fix.

**Comment by petrufm** [comment]
Unfortunately I do not have a test case :(. The potential problem has been detected directly at the source code level via static analysis.

**Comment by sezero** [comment]
Looks this is a bug introduced by commit 4c33af41ec back in 2006.
